pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -295,8 +306,10 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		}
 
 		if _, present := node.Annotations[nodeconfig.VersionAnnotation]; present {
```

Comment:
Shouldn`t we also be checking for `PubKeyHashAnnotation` here ?

Reply:
Thats not necessary, as this line is a check to see if the node is marked as fully configured or not

Reply:
I wanted to keep the node deletion in one spot so that we arent duplicating code. I also think its better if the Machine controller handles the Machine deletion logic

Reply:
> Aren't we already in machine controller?

Sorry yeah, I misread githubs formatting of the comments. 

I'm not fully sure what you are asking, are you saying that 
```

	// Update the signer with the current privateKey
	r.signer, err = signer.Create(privateKey)
	if err != nil {
		return reconcile.Result{}, errors.Wrap(err, "error creating signer")
	}
	// validate userData secret
	if err := r.validateUserData(privateKey); err != nil {
		return reconcile.Result{}, errors.Wrapf(err, "error validating userData secret")
	}
```
Should be done where it was before?

Reply:
If so I have just done so

Reply:
> Should be done where it was before?

Yes but I think that partly solves the problem.

Let me ask in a different way, what is the chance the version annotation is not present on the node object and publicKey annotation has changed.

Reply:
The version annotation not being present means that the node has not been fully configured. We have to check that the version annotation is set before checking the public key hash